pkg/reconciler/revision/resources/queue.go

```diff
@@ -225,22 +225,27 @@ func makeQueueContainer(rev *v1.Revision, cfg *config.Config) (*corev1.Container
 		// execprobe would have used (which will then check the user container).
 		// Unlike the StartupProbe, we don't need to override any of the other settings
 		// except period here. See below.
-		httpProbe = container.ReadinessProbe.DeepCopy()
-		httpProbe.Handler = corev1.Handler{
-			HTTPGet: &corev1.HTTPGetAction{
-				Port: intstr.FromInt(int(servingPort.ContainerPort)),
-				HTTPHeaders: []corev1.HTTPHeader{{
-					Name:  network.ProbeHeaderName,
-					Value: queue.Name,
-				}},
-			},
-		}
-
-		// Default PeriodSeconds to 1 if not set to make for the quickest possible startup
-		// time.
-		// TODO(#10973): Remove this once we're on K8s 1.21
-		if httpProbe.PeriodSeconds == 0 {
-			httpProbe.PeriodSeconds = 1
+		// Only add probe if Container-Freezer is not enabled so as not to continuously
+		// probe a frozen container
+		if cfg.Deployment.ConcurrencyStateEndpoint == "" {
```

Comment:
thinking out loud: I wonder if instead of disabling this just when ConcurrencyStateEndpoint is passed (which seems a somewhat unexpected side-effect, especially since so far we've avoided specifying that the -Endpoint has to freeze as opposed to log or throttle or whatever), we could instead use a StartupProbe rather than a ReadinessProbe for the default case. That is, if a user didn't specify a probe at all so Knative is adding a default probe for the user, perhaps instead of adding a readiness TCP probe, we can add a _startup_ TCP probe. Polling TCP after startup is pretty pointless - TCP does not tend to stop responding after startup, and I don't think we've ever made any guarantees that the probe we default if none is specified is a readiness probe rather than startup..

Reply:
cc @markusthoemmes @dprotaso for thoughts on that, but it seems reasonable to me that we don't default in a tcp readiness probe just to afford startup flakes now startupprobes exist..

Reply:
Anecdotally, using @julz suggestion of a startup probe and no readiness probe works nicely for the freezer.

Startup probe I used for reference: 
```json
"startupProbe": {
    "failureThreshold": 3,
    "httpGet": {
      "httpHeaders": [
        {
          "name": "K-Network-Probe",
          "value": "queue"
        }
      ],
      "path": "/",
      "port": 8012,
      "scheme": "HTTP"
    },
```